src/ConnectedMode/ServerSentEvents/SSESessionManager.cs

```diff
@@ -67,25 +71,49 @@ public void Dispose()
 
         private void SolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs activeSolutionBindingEventArgs)
         {
-            EndCurrentSession();
-
-            var bindingConfiguration = activeSolutionBindingEventArgs.Configuration;
-            var isInConnectedMode = !bindingConfiguration.Equals(BindingConfiguration.Standalone);
+            CreateSessionIfInConnectedMode(activeSolutionBindingEventArgs.Configuration);
+        }
 
-            if (!isInConnectedMode)
+        private void CreateSessionIfInConnectedMode(BindingConfiguration bindingConfiguration)
+        {
+            ISSESession sessionToLaunch;
+            lock (syncRoot)
             {
-                return;
-            }
+                if (bindingConfiguration.Equals(currentBindingConfiguration))
+                {
+                    return;
+                }
+
+                EndCurrentSession();
+
+                currentBindingConfiguration = bindingConfiguration;
+                var isInConnectedMode = !bindingConfiguration.Equals(BindingConfiguration.Standalone);
 
-            currentSession = sseSessionFactory.Create(bindingConfiguration.Project.ProjectKey);
+                if (!isInConnectedMode)
+                {
+                    return;
+                }
 
-            currentSession.PumpAllAsync().Forget();
+                currentSession = sessionToLaunch = sseSessionFactory.Create(bindingConfiguration.Project.ProjectKey);
+            }
+
+            try
+            {
+                sessionToLaunch.PumpAllAsync().Forget();
+            }
+            catch (ObjectDisposedException)
+            {
+                // ignore
```

Comment:
@rita-sonarsource I've only just realized that because the manager doesn't await the Task, it doesn't need the catch